### PR TITLE
WIP: New Slack actions

### PR DESCRIPTION
**Pull request by DoriftoShoes**
This PR adds new Slack actions to support automating user invites.
#### Actions:
- send_invite
- auth_test
- list_users
- list_channels

### CODE REVIEW

File: packs/slack/actions/auth_test.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            'token': token
+        }
+
+        data = urllib.urlencode(body)
+        response = requests.get(url=url,
+                                headers=headers, params=data)
+
+        results = response.json()
+
+        if results['ok'] is True:
```
Comment (lakshmi-kannan) [inline packs/slack/actions/auth_test.py]: `is True` is not needed.

File: packs/slack/actions/auth_test.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            return results
+        else:
+            failure_reason = ('Failed to authenticate: %s \
+                              (status code: %s)' % (response.text,
+                              response.status_code))
+            self.logger.exception(failure_reason)
+            raise Exception(failure_reason)
```
Comment (lakshmi-kannan) [inline packs/slack/actions/auth_test.py]: heh dude. you should really use `return early` approach.

File: packs/slack/config.yaml [hunk 3]
```diff
@@ -1,4 +1,26 @@
 ---
+# Used for auth test action
+auth_test_action:
+  token: ""
+
+# Used for list channels
+list_channels_action:
```
Comment (Kami) [inline packs/slack/config.yaml]: To me it kinda seems that most of those configuration options (except the token) should really just be action parameters.

Comment (Kami) [inline packs/slack/config.yaml]: Edit: I see user can also specify it as a parameter and those values there work more as a default value. I think I'm fine with that.

File: packs/slack/actions/list_channels.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  name: "list_channels"
+  runner_type: "run-python"
+  description: "List channels in a Slack Org"
+  enabled: true
+  entry_point: "list_channels.py"
+  parameters:
+    token:
+      type: "string"
+      description: "Slack API token."
+    exclude_archived:
```
Comment (Kami) [inline packs/slack/actions/list_channels.yaml]: This should be a boolean and we should send a correct integer inside the action.

File: packs/slack/actions/send_invite.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  parameters:
+    email:
+      type: "string"
+      description: "Email address to send invitation to."
+      required: true
+    first_name:
+      type: "string"
+      description: "Users first name"
+    channels:
+      type: "string"
```
Comment (Kami) [inline packs/slack/actions/send_invite.yaml]: This should be an array and we should build a comma delimited string inside an action.

Comment (Kami) [inline packs/slack/actions/send_invite.yaml]: It actually looks like you expect an array from a config, but not from a parameter :)

**Comment by DoriftoShoes** [comment]
I'm closing this PR as I moved these changes over to a new repo with auto generated actions.

https://github.com/StackStorm/st2-slack

**Pull request closed without merging**